pkg/tags/tags.go

```diff
@@ -49,21 +49,42 @@ type Tags struct {
 	tags       *sync.Map
 	stateStore storage.StateStorer
 	logger     logging.Logger
+	rand       *rand.Rand
+	randM      sync.Mutex
 }
 
 // NewTags creates a tags object
 func NewTags(stateStore storage.StateStorer, logger logging.Logger) *Tags {
+
 	return &Tags{
 		tags:       &sync.Map{},
 		stateStore: stateStore,
 		logger:     logger,
+		rand:       rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
this is the same as it was before just that it has been moved a few lines :) the idea was that the `Source` can be passed to the constructor as a parameter, allowing you to create two `NewTags` with different sources on the test level, avoid the needs for sleeps